### PR TITLE
pkg/asset/installconfig/aws: getCredentials profile handling

### LINKED ISSUE #1139: Install fails if .aws/credentials exist and profile is not used
# Version

```console
$ openshift-install version
openshift-install unreleased-master-153-g9e263e05896508752f185ec88c9592c66fdaf694
```

# Platform (aws|libvirt|openstack):

aws

# What happened?

When an `.aws/credentials` file exists and the `AWS_PROFILE` env var is not set the installer prompts for credentials but then fails due to the existence of `.aws/credentials`.  

```
[pweil@pweil-fedora installer]$ unset AWS_PROFILE
[pweil@pweil-fedora installer]$ openshift-install --dir ~/codebase/clusters/pweil-test create cluster
? SSH Public Key /home/pweil/.ssh/id_rsa.pub
? Platform aws
? AWS Access Key ID <dummy>
? AWS Secret Access Key [? for help] ****
INFO Writing AWS credentials to "/home/pweil/.aws/credentials" (https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html) 
FATAL failed to fetch Terraform Variables: failed to fetch dependency of "Terraform Variables": failed to fetch dependency of "Install Config": failed to fetch dependency of "Base Domain": failed to generate asset "Platform": open /home/pweil/.aws/credentials: file exists 


[pweil@pweil-fedora installer]$ export AWS_PROFILE=openshift-dev
[pweil@pweil-fedora installer]$ openshift-install --dir ~/codebase/clusters/pweil-test create cluster
? SSH Public Key /home/pweil/.ssh/id_rsa.pub
? Platform aws
? Region us-west-1
? Base Domain devcluster.openshift.com
? Cluster Name pweil-test
? Pull Secret [? for help] *********************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
INFO Creating cluster...                          

```
If `AWS_PROFILE` is set cluster creation continues

# What you expected to happen?

Do not fail due to existence of `.aws/credentials`.  Either append or use the credentials supplied via environment vars or some other mechanism

### PR DESCRIPTION
**Pull request by wking**
This pull requet adds support for credential-writing when the user's selected profile (as declared via `AWS_PROFILE`, falling back to `default`) does not resolve to usable credentials.

Previously, we assumed that users without a credentials file would not set `AWS_PROFILE` and so we'd hard-coded `default` as the selected profile (why select among multiple profiles when you have none?).  But that turned out not to be the case, with some users setting `AWS_PROFILE` to other values despite not having a credentials file at all.  With this pull-request, we start respecting `AWS_PROFILE` in `getCredentials`.

We also errored out when the selected profile had invalid credentials (possibly because it was undefined in the credentials file) and a credentials file already existed.  That was the safe approach, but it lead to some user confusion (#1139).  With this pull request, we use a new INI library to update the selected profile in the existing credentials file.  I'm using a new INI library, because `github.com/aws` contains an internal INI reader, but no writer logic.  Based on my limited testing, I think the new library will round-trip existing INI content successfully.  It preserves comment characters for existing comments (although new comments will use semicolons).  It seems to remove empty lines, but that is probably less of a useability burden than the previous "installer errors out" for those cases.  However, the credentials file is pretty important, so I'm personally fine dropping the two commits behind this change and continuing to error out if the file already exists (possibly with a customized error message to try and reduce user confusion).

Fixes #1139.

### CODE REVIEW

**Comment by wking** [comment]
[images][1]:

```
2019/02/01 08:52:11 Container artifacts in pod release-latest completed successfully
2019/02/01 08:52:22 error: unable to signal to artifacts container to terminate in pod release-latest, triggering deletion: could not run remote command: unable to upgrade connection: container not found ("artifacts")
2019/02/01 08:52:28 Ran for 24m51s
error: could not run steps: test "release-latest" failed: pod release-latest was already deleted
```

/test images

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/1173/pull-ci-openshift-installer-master-images/2451/build-log.txt

File: pkg/asset/installconfig/aws/aws.go [hunk 1]
```diff
@@ -133,37 +132,46 @@ func getCredentials() error {
...
 
-	return tmpl.Execute(file, map[string]string{
-		"KeyID":     keyID,
-		"Profile":   profile,
-		"SecretKey": secretKey,
-	})
+	creds.Section(profile).Key("aws_access_key_id").SetValue(keyID)
+	creds.Section(profile).Key("aws_secret_access_key").SetValue(secretKey)
+
+	tempPath := path + ".tmp"
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/aws/aws.go]: can you add to the commit comment why we are doing this. will be helpful.

File: pkg/asset/installconfig/aws/aws.go [hunk 2]
```diff
@@ -133,37 +132,46 @@ func getCredentials() error {
...
+		if os.IsNotExist(err) {
+			creds = ini.Empty()
+			creds.Section("").Comment = "https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html"
+		} else {
+			return err
+		}
 	}
-	defer file.Close()
 
 	profile := os.Getenv("AWS_PROFILE")
```
Comment (abhinavdahiya) [inline pkg/asset/installconfig/aws/aws.go]: Getenv vs LookupEnv ?

Comment (wking) [reply pkg/asset/installconfig/aws/aws.go]: > Getenv vs LookupEnv ?

I think conflating unset d ser-empty is appropriate here, but I can check and see how the AWS loaders handle set-enpty if you like.

File: pkg/asset/installconfig/aws/aws.go (hunk 1 above)
Comment (wking) [reply pkg/asset/installconfig/aws/aws.go]: > can you add to the commit comment why we are doing this.

Done with ed4b050 -> 56e22fa (e167ad3e88b has the updated commit message), which also rebases onto master.

**Pull request merged**